src/node.cc

```diff
@@ -5014,6 +5023,11 @@ int Start(int argc, char** argv) {
       TTDFlagWarning_Cond(s_ttoptReplayUri != nullptr,
           "Must set replay source info when replaying.\n");
   }
+
+  if (!envDoRecord && s_doTTRecord) {
```

Comment:
It seems like this is a function of just s_doTTRecord and not envDoRecord, right? As in, if s_doTTRecord is set, we always want to set the environment variable, regardless of what its set to?